catroid/src/main/java/org/catrobat/catroid/embroidery/EmbroideryManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.embroidery;
+
+import android.content.res.Resources;
+import android.graphics.PointF;
+import android.util.DisplayMetrics;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.common.BrickValues;
+import org.catrobat.catroid.content.XmlHeader;
+
+import java.util.ArrayList;
+
+public final class EmbroideryManager {
+	private static ArrayList<PointF> stitchPoints = new ArrayList<>();
```

Comment:
this isnt really what i meant with if you could solve this without a singleton.

anyways. id rather have a singleton than this static thingy.
why:
a utility class that is never instantiated, and only has static methods, should NEVER hava a state. (state in this case meaning that it holds data, like this one here)

the problem with the state is getting further problematic here because of calculateScreenRatio():
stitchSize is public static, so its there from the beginning. beginning here is most probably when any of the static methods are called for the first time, but still this isnt nice because in a static member initialization you access system stuff and the ProjectManager. so if youd like to unit test this, and just add a stitch point, this will blow up because theres no current project....


if youd move the calculate screen ration out of here (because its never used inside here, but only by the actor), you suddenly could make this a simple class thats only hodling the stitch data, that could might as well be instanciated wherever you please, and even extend ArrayList instead of holding one, since the only thing this does differently than a regular array is to check if the added point is a duplicate of the previous one).

if you have the calculateScreenRatio in lets say, stageListener, you can hand it over to the actor on creation, that will also save you several calls in actor to the manager.

is this specified behaviour that multiple sprites draw into one array of stitch points? (in contrast to pen where every sprite draws its own path) because if youd reduce the embroideryManager to just an extension of array list, it could be also just a member of the sprite it belongs to (since you already have a reference to the sprite in the action anyways)

i sugest we talk about this in person, draw this up on the white board and try to find a proper architecture for this.